File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.example.myapplication.databinding.ActivityMainBinding;


public class MainActivity extends AppCompatActivity implements AuthView {

    public static final String MESSAGE = "MESSAGE";

    static {
        // Used to load the 'myapplication' library on application startup.
        System.loadLibrary("myapplication");
        //System.loadLibrary("opencv_java4");
    }

    private ActivityMainBinding binding;
    private AuthPresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        mPresenter = new AuthPresenter(this);
        mPresenter.init();

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
    }

    public void onEnterPressed(View view) {
        EditText editText = binding.editName;
        EditText editPassword = binding.editPassword;
        String login = editText.getText().toString();
        String password = editPassword.getText().toString();
        mPresenter.tryLogIn(login, password);
    }

    /**
     * A native method that is implemented by the 'myapplication' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();

    @Override
    public void openContentScreen(@NonNull String welcomeMSG) {
        //Intent intent = new Intent(this, SecondActivity.class);
        //intent.putExtra(MESSAGE, welcomeMSG);
        Intent intent = new Intent(this, CameraCapture.class);
        startActivity(intent);
    }

    @Override
    public void showLoginError() {
        FragmentManager manager = getSupportFragmentManager();
        AlertFragment alert = new AlertFragment("wrong login");
        alert.show(manager, "alert");
    }

    @Override
    public void showPasswordError() {
        FragmentManager manager = getSupportFragmentManager();
        AlertFragment alert = new AlertFragment("wrong password");
        alert.show(manager, "alert");
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.WindowManager
import androidx.navigation.NavController
import androidx.navigation.findNavController
import com.example.myapplication.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    companion object {
        const val MESSAGE = "MESSAGE"

        init {
            // opencv lib binding is executed in CameraCapture activity over special opencv initializer
            // System.loadLibrary("opencv_java4");
        }
    }

    private lateinit var binding: ActivityMainBinding
    private lateinit var navController: NavController
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        navController = findNavController(R.id.nav_host_fragment_content_main)
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp()
    }
}